Sources/Decoder.swift

```diff
@@ -157,6 +157,9 @@ public struct Decoder {
                 
                 for subJSON in jsonArray {
                     guard let model = T(json: subJSON) else {
+                        if T.self is ArrayPartiallyDecodable.Type {
```

Comment:
@hkellaway when you apply the `PartiallyDecodable` logic to a user defined type it's applied only to the types you need to, if you apply that to primitive types then ALL types will become `PartiallyDecodable` which is not the intended functionality, the user must indicate what needs to be Partially or not, that's why I wrote the PartiallyDecodableBool wrapper example.